hazelcast/src/main/java/com/hazelcast/map/impl/query/MapLocalQueryRunner.java

```diff
@@ -0,0 +1,317 @@
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.query;
+
+import com.hazelcast.config.CacheDeserializedValues;
+import com.hazelcast.internal.cluster.ClusterService;
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.map.impl.LocalMapStatsProvider;
+import com.hazelcast.map.impl.MapContainer;
+import com.hazelcast.map.impl.MapServiceContext;
+import com.hazelcast.map.impl.PartitionContainer;
+import com.hazelcast.map.impl.record.Record;
+import com.hazelcast.map.impl.record.Records;
+import com.hazelcast.monitor.impl.LocalMapStatsImpl;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.query.PagingPredicate;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.impl.CachedQueryEntry;
+import com.hazelcast.query.impl.QueryableEntry;
+import com.hazelcast.query.impl.getters.Extractors;
+import com.hazelcast.query.impl.predicates.QueryOptimizer;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.OperationService;
+import com.hazelcast.spi.exception.RetryableHazelcastException;
+import com.hazelcast.spi.partition.IPartitionService;
+import com.hazelcast.util.Clock;
+import com.hazelcast.util.IterationType;
+import com.hazelcast.util.executor.ManagedExecutorService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static com.hazelcast.map.impl.query.MapQueryEngineUtils.newQueryResult;
+import static com.hazelcast.map.impl.query.MapQueryEngineUtils.newQueryResultForSinglePartition;
+import static com.hazelcast.query.PagingPredicateAccessor.getNearestAnchorEntry;
+import static com.hazelcast.spi.ExecutionService.QUERY_EXECUTOR;
+import static com.hazelcast.util.SortingUtil.compareAnchor;
+import static com.hazelcast.util.SortingUtil.getSortedSubList;
+
+/**
+ * Runs query operations in the calling thread (thus blucking it)
+ * Query evaluation per partition is run in a sequential fashion.
+ * <p>
+ * Used by query operations only: QueryOperation & QueryPartitionOperation
+ * Should not be used by proxies or any other query related objects.
+ */
+public class MapLocalQueryRunner {
+
+    protected final MapServiceContext mapServiceContext;
+    protected final NodeEngine nodeEngine;
+    protected final ILogger logger;
+    protected final QueryResultSizeLimiter queryResultSizeLimiter;
+    protected final InternalSerializationService serializationService;
+    protected final IPartitionService partitionService;
+    protected final QueryOptimizer queryOptimizer;
+    protected final OperationService operationService;
+    protected final ClusterService clusterService;
+    protected final LocalMapStatsProvider localMapStatsProvider;
+    protected final ManagedExecutorService executor;
```

Comment:
Perhaps I'm missing something, but why does this class have an executor which is only used by its subclass?